app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java

```diff
@@ -105,14 +116,60 @@ public void onMapReady(GoogleMap googleMap) {
     mMap.setMyLocationEnabled(true);
     mMap.setInfoWindowAdapter(this);
     mMap.setOnInfoWindowClickListener(this);
-    try { // TODO: might want to move this to onCreateView (not sure if listeners are duplicated)
+    mMap.getUiSettings().setZoomControlsEnabled(true);
+    mMap.setPadding(0, 0, 0, 140);
+    mLocation = DependencyFactory.getCurrentGpsTracker(getContext()).getLocation();
+    mMap.setOnMapLongClickListener(new LongClick());
+    mMap.setOnMarkerDragListener(new MarkerDrag());
+    try { // might want to move this to onCreateView (not sure if listeners are duplicated)
+      // we should not move, because only after the map is ready, can we draw markers then
       setupNearbyFavorsListener();
-
       setupFocusedFavorListen();
     } catch (Exception e) {
       CommonTools.showSnackbar(requireView(), getString(R.string.error_database_sync));
     }
-    if (focusedFavor == null) centerViewOnMyLocation();
+    // only when the app is firstly opened, the map view is centered on my location,
+    // otherwise just return to where you left before
+    if (focusedFavor == null && firstOpenApp) {
+      centerViewOnMyLocation();
+      firstOpenApp = false;
+    }
+  }
+
+  private class MarkerDrag implements GoogleMap.OnMarkerDragListener {
+
+    @Override
+    public void onMarkerDragStart(Marker marker) {}
+
+    @Override
+    public void onMarkerDrag(Marker marker) {
+      mMap.animateCamera(
+          CameraUpdateFactory.newLatLngZoom(marker.getPosition(), mMap.getMaxZoomLevel() - 5));
+    }
+
+    @Override
+    public void onMarkerDragEnd(Marker marker) {}
+  }
+
+  private class LongClick implements GoogleMap.OnMapLongClickListener {
+    @Override
+    public void onMapLongClick(LatLng latLng) {
+      if (newMarkers.size() != 0) {
+        for (Marker m : newMarkers) m.remove();
+        newMarkers.clear();
+      }
+      FavoLocation loc = new FavoLocation(mLocation);
+      loc.setLatitude(latLng.latitude);
+      loc.setLongitude(latLng.longitude);
+      Favor newFavor =
+          new Favor(
+              "", " ", DependencyFactory.getCurrentFirebaseUser().getUid(), loc, FavorStatus.EDIT);
```

Comment:
Why these empty strings in the arguments of the constructor?

Reply:
title and description will be overwritten by the code next line.

Reply:
Where are overwritten in the next line?

Reply:
` observedFavor.updateToOther(FakeItemFactory.getFavor());`

Reply:
Oh I see now, sorry!